### PR TITLE
Sysupport admin

### PR DESCRIPTION
**Pull request by velrest**
Depends on #62

### CODE REVIEW

File: frontend/app/locales/de/translations.js [hunk 1]
```diff
@@ -126,12 +133,26 @@ export default {
...
+      project: 'Projekt',
+      projects: 'Projekte',
+      billingType: 'Verrechnungs Art',
+      orderedTime: 'Bestellte Zeit',
+      usedTime: 'Verbrauchte Zeit',
+      timeTotal: 'Insgesamte Zeit',
+      unconfirmedTime: 'Ausstehende Zeit',
+      admin: 'Admin',
+      'confirm-success': 'Bestellung akzeptiert.',
+      'confirm-deny': 'Bestellung abgelehnt.'
```
Comment (anehx) [inline frontend/app/locales/de/translations.js]: Be consistent with your naming. Either camel case or dasherize but don't mix it please

File: frontend/app/locales/en/translations.js [hunk 2]
```diff
@@ -125,12 +131,26 @@ export default {
...
+      project: 'Project',
+      projects: 'Projects',
+      billingType: 'Billing Type',
+      orderedTime: 'Time Ordered',
+      usedTime: 'Time Used',
+      timeTotal: 'Time Total',
+      unconfirmedTime: 'Needs confirmation',
+      admin: 'Admin',
+      'confirm-success': 'Order accepted.',
+      'confirm-deny': 'Order denied.'
```
Comment (anehx) [inline frontend/app/locales/en/translations.js]: Same here

File: frontend/app/models/user.js [hunk 3]
```diff
@@ -42,7 +42,11 @@ export default Model.extend({
       .map(g => g.split('-')[0])
   }),
 
+  admin: computed('groups.[]', function() {
+    return this._checkGroup(window.CustomerCenter.adminGroup)
```
Comment (anehx) [inline frontend/app/models/user.js]: Please don't use a global variable. Set some kind of config property for this

File: frontend/app/sysupport-admin/detail/route.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import Route from '@ember/routing/route'
+import RSVP from 'rsvp'
```
Comment (anehx) [inline frontend/app/sysupport-admin/detail/route.js]: `import { hash } from 'rsvp'`

File: frontend/app/sysupport-admin/index/controller.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import Controller from '@ember/controller'
+
+export default Controller.extend({
+  init() {
+    this._super(...arguments)
+    this.set('headers', [
```
Comment (anehx) [inline frontend/app/sysupport-admin/index/controller.js]: This would be a use case for a computed property just returning this array

File: frontend/app/sysupport-admin/index/controller.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  },
+
+  _sortDurations(data, { order, attr }) {
+    return data.toArray().sort((a, b) => {
+      if (order === 'asc') {
+        return a
+          .get(attr)
+          .clone()
+          .subtract(b.get(attr))
+          .asMinutes()
```
Comment (anehx) [inline frontend/app/sysupport-admin/index/controller.js]: AFAIK moment durations are compareable by default..

File: frontend/mirage/config.js [hunk 7]
```diff
@@ -77,28 +93,86 @@ export default function() {
...
-    'timed-subscription-project'
+    ({ timedSubscriptionProjects }, request) => {
+      if (request.queryParams.include) {
+        let includes = request.queryParams.include.split(',')
+        let index = includes.indexOf('billing_type')
+        if (index > -1) {
+          includes.splice(index, 1, 'billing-type')
+          request.queryParams.include = includes.join()
+        }
+      }
```
Comment (anehx) [inline frontend/mirage/config.js]: Why is this necessary?

File: frontend/mirage/config.js [hunk 8]
```diff
@@ -77,28 +93,86 @@ export default function() {
...
```
Comment (anehx) [inline frontend/mirage/config.js]: Why?

File: frontend/tests/acceptance/sysupport-admin-test.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+  hooks.beforeEach(function() {
+    this.application = null
+  })
+
+  hooks.beforeEach(async function() {
+    this.application = startApp()
+    let user = server.create('user', 'admin')
+    await authenticateSession(this.application, { data: user.id })
+  })
```
Comment (anehx) [inline frontend/tests/acceptance/sysupport-admin-test.js]: Use new style tests or refactor to only use one `beforeEach`

File: frontend/tests/acceptance/timed-subscription-project-detail-test.js [hunk 10]
```diff
@@ -1,50 +1,49 @@
...
-
-  andThen(function() {
-    click('[data-test-detail-link="0"]')
-    andThen(() => {
-      assert.equal(currentURL(), '/sysupport/projects/1')
-      assert.equal(find('[data-test-project-report]').length, 5)
-      assert.equal(find('[data-test-project-order]').length, 5)
-    })
+  hooks.beforeEach(function() {
+    this.application = null
```
Comment (anehx) [inline frontend/tests/acceptance/timed-subscription-project-detail-test.js]: Why?

File: frontend/tests/acceptance/timed-subscription-project-detail-test.js [hunk 11]
```diff
@@ -1,50 +1,49 @@
...
-    andThen(() => {
-      assert.equal(currentURL(), '/sysupport/projects/1/reload')
-      click('[data-test-project-reload="0"]')
-      andThen(() => {
-        assert.equal(currentURL(), '/sysupport/projects/1')
-        assert.equal(find('[data-test-project-order]').length, 6)
-      })
-    })
+  hooks.beforeEach(async function() {
+    this.application = startApp()
```
Comment (anehx) [inline frontend/tests/acceptance/timed-subscription-project-detail-test.js]: `setupApplicationTest(hooks)` should handle this

File: frontend/tests/integration/components/cc-sort-header/component-test.js [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+      {{/cc-sort-header}}
+    `)
+
+    assert.dom('[data-test-header-name="0"]').hasText('test')
+
+    await click('[data-test-sort]')
+    await click('[data-test-sort]')
+    await click('[data-test-sort]')
+    await click('[data-test-sort]')
+  })
```
Comment (anehx) [inline frontend/tests/integration/components/cc-sort-header/component-test.js]: Wow, weird test. Sure you can't do this in a more readable way?

File: frontend/tests/unit/mixins/route-access-mixin-test.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    let ok = true
+    let mixin = EmberObject.extend(RouteAccessMixinMixin).create({
+      session,
+      groups: ['adsy-user'],
+      transitionTo: () => {
+        ok = !ok
+      }
+    })
+    mixin.beforeModel({ abort: () => {} })
+    assert.ok(ok)
```
Comment (anehx) [inline frontend/tests/unit/mixins/route-access-mixin-test.js]: Use `assert.step` and `assert.verifySteps` for this kind of assertions

File: frontend/tests/unit/mixins/route-access-mixin-test.js [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+      })
+    )
+    let mixin = EmberObject.extend(RouteAccessMixinMixin).create({
+      session,
+      groups: ['adsy-user'],
+      transitionTo: path => {
+        assert.ok(path)
+      }
+    })
+    mixin.beforeModel({ abort: () => {} })
```
Comment (anehx) [inline frontend/tests/unit/mixins/route-access-mixin-test.js]: Same as above

File: docker-compose.yml [hunk 15]
```diff
@@ -67,30 +67,30 @@ services:
...
-      - "8082:8082"
+  #rt1:
+    #image: jbfink/rt:latest
+    #command: /opt/rt4/sbin/rt-server
+    #volumes:
+      #- ./tools/docker/rt4/etc/RT_Config.pm:/opt/rt4/etc/RT_Config.pm
+      #- ./tools/docker/rt4/etc/RT_SiteConfig.pm:/opt/rt4/etc/RT_SiteConfig.pm
+      #- ./tools/docker/rt4/var/rt4:/opt/rt4/var/rt4
+    #ports:
+      #- "8082:8082"
```
[the remark this answers is not in the document]
Comment (velrest) [reply docker-compose.yml]: :+1:

File: README.md [hunk 16]
```diff
@@ -60,3 +60,7 @@ make deploy
 
 ### LDAP
 To authenticate against LDAP, we used the [Passport LDAP authentication strategy](https://github.com/vesse/passport-ldapauth)
+
+If the ucs1 container exits with seg fault (error code 139), you need to either update to a never ucs-master image (and also update your config) or set `GRUB_CMDLINE_LINUX_DEFAULT="quiet vsyscall=emulate"` in `/etc/default/grub` and then do `grub-mkconfig -o /boot/grub/grub.cfg` and `reboot`.
+
+You could also use the prod LDAP server. Config is on prod custtomer-center.
```
[the remark this answers is not in the document]
Comment (velrest) [reply README.md]: :+1:

File: frontend/app/adapters/timed-billing-type.js [hunk 17]
```diff
@@ -0,0 +1,5 @@
+import ApplicationAdapter from './application'
+
+export default ApplicationAdapter.extend({
+  pathForType: () => 'billing-type'
```
[the remark this answers is not in the document]
Comment (velrest) [reply frontend/app/adapters/timed-billing-type.js]: :+1:

File: frontend/app/components/cc-data-table/component.js [hunk 18]
```diff
@@ -0,0 +1,71 @@
+import Component from '@ember/component'
+import { computed } from '@ember/object'
+
+export default Component.extend({
+  init() {
+    this._super(...arguments)
+    this.set('sort', null)
+    this.set('search', null)
+  },
```
[the remark this answers is not in the document]
Comment (velrest) [reply frontend/app/components/cc-data-table/component.js]: :+1:

File: frontend/app/locales/de/translations.js (hunk 1 above)
Comment (velrest) [reply frontend/app/locales/de/translations.js]: :+1:

File: frontend/mirage/config.js (hunk 7 above)
Comment (velrest) [reply frontend/mirage/config.js]: The timed backend only knows the include field `billing_type`. Mirage fails since it wants to resolve the included models and there is no frontend model called `billing_type`. So either we solve this here or edit the serializer so we can use `billig-type` in the model hook.

File: frontend/tests/acceptance/timed-subscription-project-detail-test.js (hunk 11 above)
Comment (velrest) [reply frontend/tests/acceptance/timed-subscription-project-detail-test.js]: I will upgrade to the new test style once i start migrating cc to ember v3 since currently my mirage version is 4.1 and the `startMirageApp` helper is in 4.2.

File: frontend/app/models/user.js (hunk 3 above)
Comment (velrest) [reply frontend/app/models/user.js]: :+1:

File: frontend/app/sysupport-admin/detail/route.js (hunk 4 above)
Comment (velrest) [reply frontend/app/sysupport-admin/detail/route.js]: :+1:

File: frontend/app/sysupport-admin/index/controller.js (hunk 5 above)
Comment (velrest) [reply frontend/app/sysupport-admin/index/controller.js]: :+1:

File: frontend/app/sysupport-admin/index/controller.js (hunk 6 above)
Comment (velrest) [reply frontend/app/sysupport-admin/index/controller.js]: :+1:

File: frontend/mirage/config.js (hunk 8 above)
Comment (velrest) [reply frontend/mirage/config.js]: see above

File: frontend/tests/acceptance/sysupport-admin-test.js (hunk 9 above)
Comment (velrest) [reply frontend/tests/acceptance/sysupport-admin-test.js]: :+1:

File: frontend/tests/acceptance/timed-subscription-project-detail-test.js (hunk 10 above)
Comment (velrest) [reply frontend/tests/acceptance/timed-subscription-project-detail-test.js]: :+1:

File: frontend/tests/integration/components/cc-sort-header/component-test.js (hunk 12 above)
Comment (velrest) [reply frontend/tests/integration/components/cc-sort-header/component-test.js]: :+1:

File: frontend/tests/unit/mixins/route-access-mixin-test.js (hunk 13 above)
Comment (velrest) [reply frontend/tests/unit/mixins/route-access-mixin-test.js]: :+1:

File: frontend/tests/unit/mixins/route-access-mixin-test.js (hunk 14 above)
Comment (velrest) [reply frontend/tests/unit/mixins/route-access-mixin-test.js]: :+1:

File: frontend/tests/unit/mixins/route-access-mixin-test.js [hunk 19]
```diff
@@ -7,7 +7,9 @@ import { computed } from '@ember/object'
 module('Unit | Mixin | route access mixin', function(hooks) {
   setupTest(hooks)
 
-  test('access mixin positive', function(assert) {
+  test('access mixin positive', async function(assert) {
+    assert.expect(0)
```
Comment (anehx) [inline frontend/tests/unit/mixins/route-access-mixin-test.js]: What?

Comment (velrest) [reply frontend/tests/unit/mixins/route-access-mixin-test.js]: :+1:

File: frontend/tests/unit/mixins/route-access-mixin-test.js [hunk 20]
```diff
@@ -0,0 +1,29 @@
+import EmberObject from '@ember/object'
+import RouteAccessMixinMixin from 'customer-center/mixins/route-access-mixin'
+import { module, test } from 'qunit'
+import { setupTest } from 'ember-qunit'
+//import { computed } from '@ember/object'
```
Comment (anehx) [inline frontend/tests/unit/mixins/route-access-mixin-test.js]: Please remove this so we can merge...

Comment (velrest) [reply frontend/tests/unit/mixins/route-access-mixin-test.js]: :+1:
